libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -959,7 +993,15 @@ string YulUtilFunctions::storageArrayIndexAccessFunction(ArrayType const& _type)
 
 				let data := <dataAreaFunc>(array)
 				<?multipleItemsPerSlot>
-
+					<?isBytesArray>
+						let itemsPerSlot := div(0x20, <storageBytes>)
+						slot := add(data, div(index, itemsPerSlot))
```

Comment:
Can you use a switch so we only do that for long arrays?

Reply:
Actually my intention was to move this before <?isBytesArray>. In that case it should cover other multi items per slot types